test/langtools/jdk/javadoc/doclet/testIndexInherited/TestIndexInherited.java

```diff
@@ -0,0 +1,159 @@
+ */
+public class TestIndexInherited extends JavadocTester {
+
+    /**
+     * The name of the HTML index file.
+     */
+    private static final String INDEX_FILE = "index-all.html";
+
+    /**
+     * The name of the JavaScript member search index file.
+     */
+    private static final String SEARCH_FILE = "member-search-index.js";
+
+    /**
+     * Index entries for members inherited by the subclasses.
+     */
+    private static final String[] INDEX_SUBCLASSES = {"""
+        <dt><a href="pkg1/ClassB.html#methodA()" class="member-name-link">methodA()</a> \
+        - Method in class pkg1.<a href="pkg1/ClassB.html" title="class in pkg1">ClassB</a></dt>
+        """, """
+        <dt><a href="pkg2/ClassC.html#methodA()" class="member-name-link">methodA()</a> \
+        - Method in class pkg2.<a href="pkg2/ClassC.html" title="class in pkg2">ClassC</a></dt>
+        """, """
+        <dt><a href="pkg1/ClassB.html#STRING_A" class="member-name-link">STRING_A</a> \
+        - Static variable in class pkg1.<a href="pkg1/ClassB.html" title="class in pkg1">ClassB</a></dt>
+        """, """
+        <dt><a href="pkg2/ClassC.html#STRING_A" class="member-name-link">STRING_A</a> \
+        - Static variable in class pkg2.<a href="pkg2/ClassC.html" title="class in pkg2">ClassC</a></dt>
+        """};
+
+    /**
+     * Search entries for members inherited by the subclasses.
+     */
+    private static final String[] SEARCH_SUBCLASSES = {"""
+        {"p":"pkg1","c":"ClassB","l":"methodA()"}""", """
+        {"p":"pkg2","c":"ClassC","l":"methodA()"}""", """
+        {"p":"pkg1","c":"ClassB","l":"STRING_A"}""", """
+        {"p":"pkg2","c":"ClassC","l":"STRING_A"}"""};
+
+    /**
+     * Index entries for members declared in the superclass.
+     */
+    private static final String[] INDEX_SUPERCLASS = {"""
+        <dt><a href="pkg1/ClassA.html#methodA()" class="member-name-link">methodA()</a> \
+        - Method in interface pkg1.<a href="pkg1/ClassA.html" title="interface in pkg1">ClassA</a></dt>
+        """, """
+        <dt><a href="pkg1/ClassA.html#STRING_A" class="member-name-link">STRING_A</a> \
+        - Static variable in interface pkg1.<a href="pkg1/ClassA.html" title="interface in pkg1">ClassA</a></dt>
+        """};
+
+    /**
+     * Search entries for members declared in the superclass.
+     */
+    private static final String[] SEARCH_SUPERCLASS = {"""
+        {"p":"pkg1","c":"ClassA","l":"methodA()"}""", """
+        {"p":"pkg1","c":"ClassA","l":"STRING_A"}"""};
+
+    /**
+     * The default constructor.
+     */
+    public TestIndexInherited() {
+    }
+
+    /**
+     * Runs the test methods.
+     *
+     * @param args the command-line arguments
+     * @throws Exception if an errors occurs while executing a test method
+     */
+    public static void main(String... args) throws Exception {
+        TestIndexInherited tester = new TestIndexInherited();
+        tester.runTests(m -> new Object[]{Path.of(m.getName())});
```

Comment:
These days, you can simplify this down to just `tester.runTests();`. The implication of the method will check the signature of the test methods and provide the appropriate path automatically.

Reply:
Done.